solr/core/src/java/org/apache/solr/search/DelegatingCollector.java

```diff
@@ -80,9 +80,20 @@ protected void doSetNextReader(LeafReaderContext context) throws IOException {
     leafDelegate = delegate.getLeafCollector(context);
   }
 
-  public void finish() throws IOException {
+  /**
+   * From Solr 9.4 using Lucene 9.8 onwards <code>DelegatingCollector.finish</code> clashes with the
+   * super class's <code>LeafCollector.finish</code> method. Please relocate any finishing logic
+   * into the <code>DelegatingCollector.complete</code> replacement completion method.
+   */
+  @Override
+  public final void finish() throws IOException {
+    super.finish();
+  }
```

Comment:
Speculatively added this in https://github.com/apache/solr/pull/1958/commits/f90bccf5d684e46fbd3e59450dc0875a450c503b i.e. how about retaining the API and by marking it final the build would break for any custom deriving classes?

And/Or we could in the upgrade notes somehow convey about `finish` needing to change to `complete` to avoid issues?